### PR TITLE
Update iconizer.sh for Xcode 14

### PR DESCRIPTION
**Pull request by turyunus**
Apple announced support for single-size app icons for Xcode 14 at WWDC 2022.

### CODE REVIEW

**Review by stherold — CHANGES_REQUESTED** [review]
Hey thanks for your PR 👍

I think we don't need to create the Contents.json as you do now. Let me explain how the script was intended to work before:

- This script is ONLY responsible for creating the app icons. Removing the old ones was never required since the mechanism was implemented to skip the ones that did not exist. Please remove that again. If you want to remove old ones please keep the command for this in your notes or better document it on the PR you created in the other repo.
- It was never necessary to CREATE the Contents.json. The creation is done manually, especially in our case this works very well since we just copy another Contents.json. The script is just responsible for generating the assets that are already listed in the Contents.json. Therefore please remove the JSON code again.

**Comment by turyunus** [comment]
Hey,

Thanks for your reply 🙃

Generating Contents.json and removing all files logic is removed.
Can you review the pull request again?

File: xcode/iconizer.sh [hunk 1]
```diff
@@ -1,52 +1,78 @@
...
+checkInputs() {
+  ([ $1 != "null" ] && [ $2 != "null" ]) || die "Usage: sh iconizer.sh <IMAGE_PATH.pdf> <APP_ICON_PATH.appiconset>"
+  ([ -e "$1" ] && [ ${1: -4} == ".pdf" ]) || die "Did not find file $1, expected path to an image file with .pdf extension."
+  ([ -e "$2" ] && [ ${2: -11} == ".appiconset" ]) || die "Did not find folder $2, expected path should end with .appiconset extension."
+
+  PDF_PATH=$1
+  APP_ICON_SET_PATH=$2
+  CONTENTS_PATH="$APP_ICON_SET_PATH/Contents.json"
+
+  [ -e "$CONTENTS_PATH" ] || die "Did not find $CONTENTS_PATH, expected folder which contains $CONTENTS_PATH"
```
Comment (stherold) [inline xcode/iconizer.sh:31]:
```suggestion
  [ -e "$CONTENTS_PATH" ] || die "Did not find $CONTENTS_PATH. Appiconset must contain a $CONTENTS_FILE file.
```

File: xcode/iconizer.sh [hunk 2]
```diff
@@ -1,52 +1,78 @@
...
 
-[ $# == 2 ] || die "Usage: sh iconizer.sh <file name> <path to *.appiconset>"
-[ -e "$1" ] || die "Did not find file $1, expected path to an image file."
+  IMAGES_LENGTH=$(jq ".images | length" $CONTENTS_PATH)
 
-CONTENT_FILE_PATH="$2/$CONTENTS_FILE"
+  ([ $IMAGES_LENGTH != "null" ] && [ $IMAGES_LENGTH ] >0) || die "Did not find any images in $CONTENTS_PATH"
```
Comment (stherold) [inline xcode/iconizer.sh:35]: The integer comparison seems to be wrong. Tested on console and the following works ([see also here](https://stackoverflow.com/a/18133196/971329)):

```suggestion
  ([ "$IMAGES_LENGTH" != "null" ] && (( $IMAGES_LENGTH > 0 ))) || || die "Did not find any images in $CONTENTS_PATH"
```

File: xcode/iconizer.sh [hunk 3]
```diff
@@ -1,52 +1,78 @@
...
+  echo "All Done 🎉🎉🎉"
+}
 
-[ -e "$CONTENT_FILE_PATH" ] || die "Did not find $CONTENT_FILE_PATH, expected folder which contains $CONTENTS_FILE"
+generateImage() {
+  image=$(jq ".images[$1]" $CONTENTS_PATH)
+  [ "$image" != "null" ] || break
 
-echo "Creating icons from $1 and updating $CONTENT_FILE_PATH..."
+  scale=$(echo $image | jq ".scale" | cut -d "\"" -f 2 | cut -d "x" -f 1)
```
Comment (stherold) [inline xcode/iconizer.sh:50]: Removing the quotation marks is even easier:

```suggestion
  scale=$(echo $image | jq -r ".scale" | cut -d "x" -f 1)
```

File: xcode/iconizer.sh [hunk 4]
```diff
@@ -1,52 +1,78 @@
...
+  [ $scale != "null" ] || scale=1
 
-i=0
+  sizePT=$(echo $image | jq ".size" | cut -d "\"" -f 2 | cut -d "x" -f 1)
```
Comment (stherold) [inline xcode/iconizer.sh:53]: Same here

```suggestion
  sizePT=$(echo $image | jq -r ".size" | cut -d "x" -f 1)
```

File: xcode/iconizer.sh [hunk 5]
```diff
@@ -1,52 +1,78 @@
...
+  newFileName="appicon_${sizePX}.png"
 
-while :
-do
-    image=$(jq ".images[$i]" $CONTENT_FILE_PATH)
-    scale=$(echo $image | jq ".scale" | cut -d "\"" -f 2 | cut -d "x" -f 1 )
-    sizePT=$(echo $image | jq ".size" | cut -d "\"" -f 2 | cut -d "x" -f 1 )
-    sizePX=$(bc -l <<< "scale=1; $sizePT*$scale" | cut -d "." -f 1)
-    newFileName="appicon_${sizePX}.png"
+  echo -n "Updading Contents.js for $newFileName"
```
Comment (stherold) [inline xcode/iconizer.sh:57]:
```suggestion
  echo -n "Updading $CONTENTS_FILE to $newFileName"
```

File: xcode/iconizer.sh [hunk 6]
```diff
@@ -1,52 +1,78 @@
...
+# Functions
+
+checkInputs() {
+  ([ $1 != "null" ] && [ $2 != "null" ]) || die "Usage: sh iconizer.sh <IMAGE_PATH.pdf> <APP_ICON_PATH.appiconset>"
+  ([ -e "$1" ] && [ ${1: -4} == ".pdf" ]) || die "Did not find file $1, expected path to an image file with .pdf extension."
+  ([ -e "$2" ] && [ ${2: -11} == ".appiconset" ]) || die "Did not find folder $2, expected path should end with .appiconset extension."
+
+  PDF_PATH=$1
+  APP_ICON_SET_PATH=$2
+  CONTENTS_PATH="$APP_ICON_SET_PATH/Contents.json"
```
Comment (stherold) [inline xcode/iconizer.sh:29]:
```suggestion
  CONTENTS_FILE="Contents.json"
  CONTENTS_PATH="$APP_ICON_SET_PATH/$CONTENTS_FILE"
```

**Review by stherold — CHANGES_REQUESTED** [review]
Nice 👍 I've just some minor updates. But that looks already quite cool :)

File: xcode/iconizer.sh (hunk 2 above)
Comment (turyunus) [reply xcode/iconizer.sh]: Both of them work correctly but I updated the line as per your suggestion.

File: xcode/iconizer.sh (hunk 4 above)
Comment (turyunus) [reply xcode/iconizer.sh]: done 👍

File: xcode/iconizer.sh (hunk 5 above)
Comment (turyunus) [reply xcode/iconizer.sh]: done 👍

File: xcode/iconizer.sh (hunk 6 above)
Comment (turyunus) [reply xcode/iconizer.sh]: done 👍

File: xcode/iconizer.sh (hunk 2 above)
Comment (stherold) [reply xcode/iconizer.sh]: Really ? For me it didn't work locally on the console but I don't know why 🤔 Anyway thx for the update :)

File: xcode/iconizer.sh [hunk 7]
```diff
@@ -1,52 +1,79 @@
...
+  for i in $(seq 0 $IMAGES_LENGTH); do
+    generateImage $i
+  done
+  echo "All Done 🎉🎉🎉"
+}
 
-[ -e "$CONTENT_FILE_PATH" ] || die "Did not find $CONTENT_FILE_PATH, expected folder which contains $CONTENTS_FILE"
+generateImage() {
+  image=$(jq ".images[$1]" $CONTENTS_PATH)
+  [ "$image" != "null" ] || break
```
Comment (Blackjacx) [inline xcode/iconizer.sh:49]: Here I get a complaint when I execute this script - it should be return in a function:
```
Creating icons from Targets/DBFFM/Resources/Assets.xcassets/AppIcon.appiconset and updating Targets/DBFFM/Resources/Assets.xcassets/AppIcon.appiconset/Contents.json...
Updading Contents.json to appicon_1024.png ✅
Generating appicon_1024.pngconvert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `Targets/DBFFM/Resources/Assets.xcassets/AppIcon.appiconset/appicon_1024.png' @ warning/png.c/MagickPNGWarningHandler/1750.
 ✅
---- > /dev/fd/14: line 49: break: only meaningful in a `for', `while', or `until' loop
Updading Contents.json to appicon_0.png ✅
Generating appicon_0.pngconvert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `Targets/DBFFM/Resources/Assets.xcassets/AppIcon.appiconset/appicon_0.png' @ warning/png.c/MagickPNGWarningHandler/1750.
 ✅
All Done 🎉🎉🎉
```

**Review by Blackjacx — CHANGES_REQUESTED** [review]
Thanks for your updates. Now when I run the script it complains about the `break` (see comment below). Apart from that everything works :)

File: xcode/iconizer.sh (hunk 7 above)
Comment (turyunus) [reply xcode/iconizer.sh]: This issue is fixed by updating for loop and it is more understandable now.

Comment (stherold) [reply xcode/iconizer.sh]: Awesome. Thanks for the contribution 🔥

**Review by stherold — APPROVED** [review]
Awesome thanks for the contribution 🔥

**Review by Blackjacx — APPROVED** [review]

**Review by Blackjacx — APPROVED** [review]

**Pull request merged**